### ISSUE TITLE
Bug in IdentifyWait

### ISSUE THREAD

**Issue by vyzo** [libp2p/go-libp2p#2981]
<details>
<summary>Version Information</summary>
<pre>
	github.com/libp2p/go-libp2p v0.36.4
</pre>
</details>

The code in `IdentifyWait`
```
	e, found := ids.conns[c]
	if !found {
		// No entry found. We may have gotten an out of order notification. Check it we should have this conn (because we're still connected)
		// We hold the ids.connsMu lock so this is safe since a disconnect event will be processed later if we are connected.
		if c.IsClosed() {
			log.Debugw("connection not found in identify service", "peer", c.RemotePeer())
			ch := make(chan struct{})
			close(ch)
			return ch
		} else {
			ids.addConnWithLock(c)
		}
	}
```

and even if `!found` we are working with `e` below, which leads to an identify deadlock. Patch forthcoming

**Comment by vyzo** [comment]
also, the thing needs to be a pointer, which is fixed in #2982.

**Comment by vyzo** [comment]
no bug, we are copying the entry; sorry.
